src/data/finset.lean

```diff
@@ -2551,6 +2551,44 @@ lemma disjoint_iff_disjoint_coe {α : Type*} {a b : finset α} [decidable_eq α]
   disjoint a b ↔ disjoint (↑a : set α) (↑b : set α) :=
 by { rw [finset.disjoint_left, set.disjoint_left], refl }
 
+/--
+Given a set A and a set B inside it, we can shrink A to any appropriate size, and keep B
+inside it.
+-/
+lemma exists_intermediate_set {A B : finset α} (i : ℕ)
+  (h₁ : card A ≥ i + card B) (h₂ : B ⊆ A) :
+  ∃ (C : finset α), B ⊆ C ∧ C ⊆ A ∧ card C = i + card B :=
+begin
+  rcases nat.le.dest h₁ with ⟨k, _⟩,
+  clear h₁,
+  induction k with k ih generalizing A,
+  { exact ⟨A, h₂, subset.refl _, h.symm⟩ },
+  { have: (A \ B).nonempty,
```

Comment:
```suggestion
  { have : (A \ B).nonempty,
```
https://github.com/leanprover-community/mathlib/blob/master/docs/contribute/style.md#structuring-definitions-and-theorems